openpype/hosts/maya/plugins/publish/extract_arnold_scene_source.py

```diff
@@ -113,34 +111,62 @@ def process(self, instance):
         instance.data["representations"].append(representation)
 
     def _extract(self, nodes, attribute_data, kwargs):
-        self.log.info("Writing: " + kwargs["filename"])
+        self.log.info(
+            "Writing {} with:\n{}".format(kwargs["filename"], kwargs)
+        )
         filenames = []
         # Duplicating nodes so they are direct children of the world. This
         # makes the hierarchy of any exported ass file the same.
-        with delete_after() as delete_bin:
+        with lib.delete_after() as delete_bin:
             duplicate_nodes = []
             for node in nodes:
+                # Only interested in transforms:
+                if cmds.nodeType(node) != "transform":
+                    continue
+
+                # Only interested in transforms with shapes.
+                shapes = cmds.listRelatives(
+                    node, shapes=True, fullPath=True
+                ) or []
+                if not shapes:
+                    continue
+
+                parent = cmds.listRelatives(
+                    node, parent=True, fullPath=True
+                )[0]
                 duplicate_transform = cmds.duplicate(node)[0]
-
-                # Discard the children.
-                shapes = cmds.listRelatives(duplicate_transform, shapes=True)
-                children = cmds.listRelatives(
-                    duplicate_transform, children=True
+                duplicate_transform = "{}|{}".format(
+                    parent, duplicate_transform
                 )
-                cmds.delete(set(children) - set(shapes))
 
                 duplicate_transform = cmds.parent(
                     duplicate_transform, world=True
                 )[0]
```

Comment:
This will fail if the original transform was already a root object (did not have a parent) so we should check for that.
```suggestion

                if cmds.listRelatives(duplicate_transform, parent=True):
                    duplicate_transform = cmds.parent(
                        duplicate_transform, world=True
                    )[0]
```